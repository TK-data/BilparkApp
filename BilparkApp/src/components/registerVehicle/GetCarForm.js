import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Dimensions } from 'react-native';
import { Button, Text, View, H2, Spinner, Icon } from 'native-base';
import CarForm from './CarForm';
import { declineCar, acceptCar } from '../../actions/registerCar';

class GetCarForm extends Component {

  render() {

    let main = (
      <CarForm />
    );

    if (this.props.carSaved && this.props.carSaved !== 'null') {

      const car = JSON.parse(this.props.carSaved);
      main = (
        <View style={styles.content}>
          <H2 style={styles.header}>DIN BIL:</H2>
          <View style={styles.content}>
            <View style={styles.carContent}>
              <View style={styles.textContainer}>
                <Text style={styles.text}>REGNR:</Text>
                <Text style={styles.text}>{car.RegNr}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>MERKE:</Text>
                <Text style={styles.text}>{car.Brand}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>MODELL:</Text>
                <Text style={styles.text}>{car.Model}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>REGISTRERINGSÅR:</Text>
                <Text style={styles.text}>{car.RegYear}</Text>
              </View>
            </View>
          </View>
        </View>
      );
    } else if (this.props.isAccepted) {
      const car = JSON.parse(this.props.carFetched);
      main = (
        <View style={styles.content}>
          <H2 style={styles.header}>DIN BIL:</H2>
          <View style={styles.content}>
            <View style={styles.carContent}>
              <View style={styles.textContainer}>
                <Text style={styles.text}>REGNR:</Text>
                <Text style={styles.text}>{car.RegNr}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>MERKE:</Text>
                <Text style={styles.text}>{car.Brand}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>MODELL:</Text>
                <Text style={styles.text}>{car.Model}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>REGISTRERINGSÅR:</Text>
                <Text style={styles.text}>{car.RegYear}</Text>
              </View>
            </View>
          </View>
        </View>
      );
    } else if (this.props.isLoading) {
      main = (
        <View>
          <Spinner color="white" />
        </View>
      );
    } else if (this.props.carFetched) {
      const car = JSON.parse(this.props.carFetched);

      main = (
        <View style={styles.content}>
          <Text style={styles.header}>ER DETTE DIN BIL?</Text>
          <View style={styles.content}>
            <View style={styles.carContent}>
              <View style={styles.textContainer}>
                <Text style={styles.text}>REGNR:</Text>
                <Text style={styles.text}>{car.RegNr}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>MERKE:</Text>
                <Text style={styles.text}>{car.Brand}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>MODELL:</Text>
                <Text style={styles.text}>{car.Model}</Text>
              </View>
              <View style={styles.textContainer}>
                <Text style={styles.text}>REGISTRERINGSÅR:</Text>
                <Text style={styles.text}>{car.RegYear}</Text>
              </View>
            </View>
          </View>
          <View style={styles.buttonContent}>
            <Button
              bordered
              light
              onPress={() => {
                this.props.acceptCar(this.props.carFetched);
              }}
              style={styles.button}
            >
              <Icon name="md-checkmark" style={{ fontSize: 40 }} />
            </Button>
            <Button
              bordered
              light
              onPress={() => {
                this.props.declineCar();
              }}
              style={styles.button}
            >
              <Icon name="md-close" style={{ fontSize: 40 }} />
            </Button>
          </View>
        </View>
      );
    }

    return (
      <View style={styles.content}>
        {main}
      </View>
    );
  }
}

const width = Dimensions.get('window').width;

const styles = StyleSheet.create({
  searchContainer: {
    flex: 1,
    alignItems: 'center',
  },
  searchButton: {
    alignSelf: 'center',
  },
  buttonContent: {
    flex: 1,
    flexDirection: 'row-reverse',
    justifyContent: 'space-around',
  },
  carContent: {
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 2,
  },
  content: {
    flex: 1,
  },
  button: {
    alignSelf: 'center',
  },
  card: {
    width: (width - 20),
  },
  header: {
    textAlign: 'center',
    color: 'white',
    marginBottom: 10,
  },
  textContainer: {
    width: (width - 40),
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 10,
  },
  text: {
    color: 'white',
  },
});

const mapStateToProps = (state) => {

  return {
    carSaved: state.auth.car,
    carFetched: state.carFetch.car,
    isLoading: state.carFetch.isLoading,
    hasErrored: state.carFetch.hasErrored,
    isAccepted: state.carFetch.isAccepted,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    declineCar: () => dispatch(declineCar()),
    acceptCar: car => dispatch(acceptCar(car)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(GetCarForm);
