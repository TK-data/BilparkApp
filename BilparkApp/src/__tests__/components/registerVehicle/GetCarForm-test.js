import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import GetCarForm from '../../../components/registerVehicle/GetCarForm';

const initialState = {
  hasErrored: false,
  isLoading: true,
  auth: {
    hasErrored: false,
    isLoading: false,
    user: 'user',
  },
  carFetch: {
    hasErrored: false,
    isLoading: false,
    car: '',
    isAccepted: false,
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing GetCarForm', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <GetCarForm />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
