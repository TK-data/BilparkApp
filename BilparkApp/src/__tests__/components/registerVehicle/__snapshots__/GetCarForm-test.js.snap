// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GetCarForm Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GetCarForm
    acceptCar={[Function]}
    carFetched=""
    carSaved={undefined}
    declineCar={[Function]}
    hasErrored={false}
    isAccepted={false}
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(CarForm) />,
      "style": Object {
        "flex": 1,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(CarForm) />,
        "style": Object {
          "flex": 1,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
